### ISSUE TITLE
IndexError: invalid index of a 0-dim tensor

### ISSUE THREAD

**Issue by zjykzj** [xiongzihua/pytorch-YOLO-v1#43]
Met this question when using `predict.py`. Error code location is located at

```
        i = order[0]
        keep.append(i)
```

after check the code, I found the key is to update `squeeze()` use

```
        # before
        ids = (ovr<=threshold).nonzero().squeeze()
        # after
        ids = (ovr<=threshold).nonzero().squeeze(-1)
```

this change can solve the problem, when `ids` shape == (1,1), then squeeze(-1) return (1)

**Comment by AlbertMa123** [comment]
应该是pytorch版本升级导致的问题，最后一次order，order=tensor(25)（例如），此时shape=[]，即不能用order[0]来获取第一个元素，建议这么写：
if order.dim()==0:
order = order.reshape(-1)

**Comment by zjykzj** [comment]
> 应该是pytorch版本升级导致的问题，最后一次order，order=tensor(25)（例如），此时shape=[]，即不能用order[0]来获取第一个元素，建议这么写： if order.dim()==0: order = order.reshape(-1)

Yes! YOLOv1 is a product of about 10 years ago, so there will be many changes in the current software environment. I think most people will pay attention to the open-source works that have implemented algorithms before in order to better delve into the field of computer vision. I have also tried to implement some YOLO series algorithms, and those interested can check them out:

* [zjykzj](https://github.com/zjykzj)/[YOLO5Face](https://github.com/zjykzj/YOLO5Face)
* [zjykzj](https://github.com/zjykzj)/[YOLOv3](https://github.com/zjykzj/YOLOv3)
* [zjykzj](https://github.com/zjykzj)/[YOLOv2](https://github.com/zjykzj/YOLOv2)
* [zjykzj](https://github.com/zjykzj)/[YOLOv1](https://github.com/zjykzj/YOLOv1)
